src/zope/testrunner/runner.py

```diff
@@ -533,6 +573,14 @@ def spawn_layer_in_subprocess(result, script_parts, options, features,
 
     finally:
         result.done = True
+        if child is not None:
+            # Regardless of whether the process ran to completion, we
+            # must properly cleanup the process to avoid
+            # `ResourceWarning: subprocess XXX is still alive` and
+            # `ResourceWarning: unclosed file` for its stdout and
+            # stderr.
+            child.kill()
+            child.communicate()
```

Comment:
I would mildly prefer `child.terminate()` and a little timeout loop calling `child.poll()` before resorting to `child.kill()`, on general principles.

Reply:
The Popen documentation [suggests using kill](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate).

Reply:
Yes, after a timeout.  I also want a timeout before `.kill()` ;)

Again, no specific reason, so I won't mind if you don't do that here.

Reply:
At this point in the code, the child process *should* already be terminated. There's a loop above that's effectively polling for it to die (albeit without a timeout).

Reply:
That sounds convincing!